src/App.jsx

```diff
@@ -1,3 +1,102 @@
+import { React, useState, useEffect } from "react";
+import { BrowserRouter, Routes, Route } from "react-router-dom";
+import { Nav } from "./components/Nav.jsx";
+import { NotFoundPage } from "./pages/NotFoundPage.jsx";
+import { Home } from "./pages/Home.jsx";
+import { Popular } from "./pages/Popular.jsx";
+import { MovieDetails } from "./pages/MovieDetails.jsx";
+import { PopularMovieDetails } from "./pages/PopularMovieDetails.jsx";
+
 export const App = () => {
-  return <div>Find me in src/app.jsx!</div>;
+    //useState to save the data array from API into movieList
+    const [upcomingList, setUpcomingList] = useState([]);
+    const [popularList, setPopularList] = useState([]);
+
+    //declaring the movie API for upcoming movies
```

Comment:
If you want App.jsx a bit 'cleaner' as the entry point, maybe you could have moved parts of this syntax to the component/page where it belongs?